ccm_web/server/src/lti/lti.service.ts

```diff
@@ -58,10 +58,14 @@ export class LTIService implements BeforeApplicationShutdown {
     provider.onConnect(async (token: IdToken, req: Request, res: Response) => {
       logger.debug(`The LTI launch was successful! User info: ${JSON.stringify(token.userInfo)}`)
       const customLTIVariables = token.platformContext.custom
-      if (customLTIVariables.login_id === undefined) {
+      if (customLTIVariables.login_id === undefined || customLTIVariables.course_id === undefined ||
+        customLTIVariables.roles === undefined || customLTIVariables.is_root_account_admin === undefined) {
```

Comment:
Not sure how many more properties there will be, but if this gets much longer, we might want to think about a more succinct validation strategy, using a schema or something. This looks okay for now.

Reply:
Yeah...